macosx.gypi

```diff
@@ -22,11 +22,5 @@
             'NN_USE_EVENTFD',
             'NN_HAVE_MSG_CONTROL'
         ],
-        'include_dirs': [
-          'deps/nanomsg/src',
-        ],
-        'cflags': [
-          '-O3 -D_THREAD_SAFE'
```

Comment:
> What are the implications of removing this?

there are zero implications of removing flags that don't work in the first place.

> Why were we previously defining `_THREAD_SAFE`?

looks like this setting was introduced to the build upstream in https://github.com/nanomsg/nanomsg/commit/df5f8cd392a487071200e881af64af96d038921f#diff-851a07ee321dc42003ac94cf4a9628faR258

I have no idea why threads would ever be used by the library, there may be a use-case or some internal nanomsg mechanism unknown to me using a pthread but ostensibly, just looking at it from the outside of the implementation, IMO such use is unjustified.